evil-common.el

```diff
@@ -2143,12 +2143,15 @@ register instead of replacing its content."
     (set-register register text))))
 
 (defun evil-register-list ()
-  "Returns an alist of all registers"
+  "Returns an alist of all registers, but only those named
+with number or character. Registers with symbol or string in names are ignored
+to keep Vim compatibility with register jumps."
   (sort (append (mapcar #'(lambda (reg)
                             (cons reg (evil-get-register reg t)))
                         '(?\" ?* ?+ ?% ?# ?/ ?: ?. ?-
                               ?1 ?2 ?3 ?4 ?5 ?6 ?7 ?8 ?9))
-                register-alist nil)
+                (cl-remove-if-not #'(lambda (reg) (number-or-marker-p reg)) register-alist)
```

Comment:
What's the difference between `(lambda (reg) (number-or-marker-p reg))` and `number-or-marker-p`? Why not just use the latter?

Reply:
Also, evil supports Emacs 24.1, which does not have cl-lib bundled. :-/

Reply:
Maybe you can do `(evil-filter-list (lambda (reg) (not (number-or-marker-p (car reg)))) (copy-list register-alist))` or something similar.

Reply:
Thanks for catching this! I was working in few iterations, so `(lambda (reg) ...)` left. Will fix that.

Regarding `cl-lib` issue, isn't already used in Evil, e.g. `evil-jumps.el`? `cl-remove-if-not` is already used there; are you planning to remove these calls?

Reply:
Ah, turns out I'm actually stupid. Evil already depends on `cl-lib`. You'll just need to add a `(require 'cl-lib)` on top of `evil-common.el` and it'll be fine.

Reply:
Done :)